test/fixtures/fixture-sharepointer.js

```diff
@@ -0,0 +1,29 @@
+module.exports = function(){
```

Comment:
This may be where I'd bring in sinon instead of writing the mock boilerplate below.

I'd write the stubs closer to the tests that need them, and only stub the methods being used during that test. e.g. something like:
```javascript
exports.testLogin = function() {
  var app = proxyquire('../../application.js', {
    sharepointer: {
      login: stub.callsArgWith(0, { session : '1a' })
    }
  });

  request(app)
    .post('/login')
    .send({username : 'foo', password : 'bar'})
    .expect(200)
    .end(function(err, res){
      assert.ok(!err);
      assert.ok(res.body.session);
      assert.equal(res.body.session, '1a');
      return done();
  });
};
```
Nothing wrong with below though as a pattern too.